src/sql/workbench/browser/designer/designer.ts

```diff
@@ -904,11 +976,25 @@ export class Designer extends Disposable implements IThemable {
 					table.registerPlugin(deleteRowColumn);
 					columns.push(deleteRowColumn.definition);
 				}
-				table.columns = columns;
+				table.columns = buttonColumn.concat(columns);
+				table.columns.forEach((v, i) => table.columns[i].id = table.columns[i].id || i.toString());
 				table.grid.onBeforeEditCell.subscribe((e, data): boolean => {
 					return data.item[data.column.field].enabled !== false;
 				});
-
+				let disabledActions = this._actions.filter(b => b.enabled === false);
```

Comment:
this doesn't seem correct, you should only touch the actions belonging to this table.